src/com/facebook/buck/cli/TestCommand.java

```diff
@@ -605,6 +609,40 @@ public int compare(TestRule o1, TestRule o2) {
     return builder.build();
   }
 
+  @VisibleForTesting
+  Iterable<TestRule> filterUntestableRules(
+      Iterable<TestRule> testRules) {
+
+    ImmutableSortedSet.Builder<TestRule> builder =
+        ImmutableSortedSet.orderedBy(
+            new Comparator<TestRule>() {
+              @Override
+              public int compare(TestRule o1, TestRule o2) {
+                return o1.getBuildTarget().getFullyQualifiedName().compareTo(
+                    o2.getBuildTarget().getFullyQualifiedName());
+              }
+            });
+
+    for (TestRule rule : testRules) {
+      // xctool doesn't run XCUITests yet, but project generation works
+      boolean isAppleUITest = false;
+      if (rule instanceof AppleTest) {
+        AppleTest appleTestRule = (AppleTest)rule;
+        isAppleUITest = appleTestRule.isUiTest();
+        if (isAppleUITest) {
+          LOG.warn("Not running rule '%s' as XCUITests aren't supported.",
+              rule.getFullyQualifiedName());
+        }
+      }
+
+      if (!isAppleUITest) {
+        builder.add(rule);
+      }
+    }
+
+    return builder.build();
+  }
```

Comment:
As far as I know, there is currently no way to run these test from the commandline except for using xcodebuild with a scheme and project. To run these tests you need to use some special testmanager daeamon and setup some more things, I'm think `xctest` is just for running logic tests. facebook/xctool#534 has some investigation into how involved running uitests is but no one seems to have reverse engineered it yet.

Letting `AppleTest` decide this instead of the `TestRunner` doesn't seem like bad idea. Maybe we should extend the `TestRule` interface with a `isSupported`/`testCanRun` flag that is used for filtering?